docs/protocol/src/crypto/tct.md

```diff
@@ -0,0 +1,383 @@
+
+At any given moment in time, the aggregate of all note commitments produced in the history of the
+chain can be thought of as a triply-nested sequence (an _eternity_) of sequences (_epochs_) of
+sequences (_blocks_) of note commitments. The tiered commitment tree represents these triply-nested
+sequences as triply-nested tiers.
+
+The entire commitment tree consists of three tiers, each of which is 8 levels tall.
+
+- The entire commitment tree consists of a top-level [_tier_](#tiers) (which we call an _eternity_),
+which contains 8 nested levels of [_nodes_](#nodes);
+- The children of the bottom-most [_nodes_](#nodes) in an eternity are [_tiers_](#tiers) (which we
+  call _epochs_), which each contain 8 nested levels of [_nodes_](#nodes).
+- The children of the bottom-most [_nodes_](#nodes) in an epoch are [_tiers_](#tiers) (which we call
+  _blocks_), which each contain 8 nested levels of [_nodes_](#nodes).
+- The children of the bottom-most [_nodes_](#nodes) in a block are [_leaves_](#leaves) that contain
+  one commitment each.
+
+```text
+Eternity┃           ╱╲ ◀───────────── Eternity Root
+    Tree┃          ╱││╲
+        ┃         * ** *           ╮
+        ┃      *   *  *   *        │ 8 levels
+        ┃   *     *    *     *     ╯
+        ┃  ╱╲    ╱╲    ╱╲    ╱╲
+        ┃ ╱││╲  ╱││╲  ╱││╲  ╱││╲ ◀─── Epoch Root
+                     ┌──┘
+                     │
+                     │
+   Epoch┃           ╱╲ ◀───────────── Epoch Root
+    Tree┃          ╱││╲
+        ┃         * ** *           ╮
+        ┃      *   *  *   *        │ 8 levels
+        ┃   *     *    *     *     ╯
+        ┃  ╱╲    ╱╲    ╱╲    ╱╲
+        ┃ ╱││╲  ╱││╲  ╱││╲  ╱││╲ ◀─── Block Root
+                 ▲
+                 └───┐
+                     │
+                     │
+   Block┃           ╱╲ ◀───────────── Block Root
+    Tree┃          ╱││╲
+        ┃         * ** *           ╮
+        ┃      *   *  *   *        │ 8 levels
+        ┃   *     *    *     *     ╯
+        ┃  ╱╲    ╱╲    ╱╲    ╱╲
+        ┃ ╱││╲  ╱││╲  ╱││╲  ╱││╲ ◀─── Block Leaf
+                                      = Note Commitment
+```
+
+As a consequence of this structure, the maximum capacity of a tier is $4^8 = 65,536$, and the
+maximum capacity of the entire tree is $4^{3 \cdot 8} = 281,474,976,710,656$.
+
+### Structural Definitions
+
+#### Tiers
+
+A tier is either _empty_ or it is a [_node_](#nodes).
+
+#### Nodes
+
+A node in a tree contains a _non-empty_ list of up to four _children_.
+
+If a node has an $n$th child (for $n > 0$), it must have an $(n - 1)$th child (there cannot be gaps
+in the list of children).
+
+Each node must be _left-to-right filled_:
+
+- if its children are nodes, all its children but its rightmost child are _perfect_, and its
+  rightmost child is left-to-right filled (where a _perfect_ node has exactly four children and if its
+  children are nodes, they are also perfect) or
+- if its children are leaves or tiers, the node is left-to-right filled.
```

Comment:
It would be clearer if this defined "leaf nodes" and "internal nodes", since conventionally, a node is either a leaf or an internal node, rather than being defined to exclude leaves.